libsrc2/hyper.c

```diff
@@ -78,10 +78,10 @@ offset_to_index(int ndims,
 
 /** The main restructuring code.
  */
-void restructure_array(int ndims,    /* Dimension count */
+static void restructure_array(hsize_t ndims,    /* Dimension count */
```

Comment:
Hi - unfortunately restructure_array() is used in one of the MINC tools, so it must remain public. If there is a warning you're seeing that this fixes, we need to find a better solution.